packages/material-ui/src/ButtonBase/TouchRipple.js

```diff
@@ -98,169 +98,191 @@ export const styles = theme => ({
   },
 });
 
-class TouchRipple extends React.PureComponent {
-  // Used to filter out mouse emulated events on mobile.
-  // ignoringMouseDown = false;
+const TouchRipple = React.forwardRef(function TouchRipple(props, ref) {
+  const { center: centerProp = false, classes, className, ...other } = props;
+  const [ripples, setRipples] = React.useState([]);
+  const nextKey = React.useRef(0);
+  const rippleCallback = React.useRef(null);
+
+  React.useEffect(() => {
+    if (rippleCallback.current) {
+      rippleCallback.current();
+      rippleCallback.current = null;
+    }
+  }, [ripples]);
 
+  // Used to filter out mouse emulated events on mobile.
+  const ignoringMouseDown = React.useRef(false);
   // We use a timer in order to only show the ripples for touch "click" like events.
   // We don't want to display the ripple for touch scroll events.
-  // startTimer = null;
+  const startTimer = React.useRef(null);
```

Comment:
Is the "established rule" to use null for reference to DOM nodes only?

```suggestion
  const startTimer = React.useRef();
```

Reply:
```diff
useRef() = { current: undefined }
createRef() = { current: null }
```

It's about introducing as little change as possible.